cca_example_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix/mat64"
+	"github.com/gonum/stat"
+)
+
+// symView is a helper for getting a View of a SymDense.
+type symView struct {
+	sym *mat64.SymDense
+
+	i, j, r, c int
+}
+
+func (s symView) Dims() (r, c int) { return s.r, s.c }
+
+func (s symView) At(i, j int) float64 {
+	if i < 0 || s.r <= i {
+		panic("i out of bounds")
+	}
+	if j < 0 || s.c <= j {
+		panic("j out of bounds")
+	}
+	return s.sym.At(s.i+i, s.j+j)
+}
+
+func (s symView) T() mat64.Matrix { return mat64.Transpose{s} }
+
+func ExampleCanonicalCorrelations() {
+	// This example is directly analogous to Example 3.5 on page 87 of
+	// Koch, Inge. Analysis of multivariate and high-dimensional data.
+	// Vol. 32. Cambridge University Press, 2013. ISBN: 9780521887939
+
+	// bostonData is the Boston Housing Data of Harrison and Rubinfeld (1978)
+	n, _ := bostonData.Dims()
+	var xd, yd = 7, 4
+	// The variables (columns) of bostonData can be partitioned into two sets:
+	// those that deal with environmental/social variables (xdata), and those
+	// that contain information regarding the individual (ydata). Because the
+	// variables can be naturally partitioned in this way, these data are
+	// appropriate for canonical correlation analysis. The columns (variables)
+	// of xdata are, in order:
+	//  per capita crime rate by town,
+	//  proportion of non-retail business acres per town,
+	//  nitric oxide concentration (parts per 10 million),
+	//  weighted distances to Boston employment centres,
+	//  index of accessibility to radial highways,
+	//  pupil-teacher ratio by town, and
+	//  proportion of blacks by town.
+	xdata := bostonData.View(0, 0, n, xd)
```

Comment:
Note gonum/matrix#416. This should be rewritten as `xdata := bostonData.Slice(0, n, 0, xd)`.